test/zdtm/static/socket-tcp-close2.c

```diff
@@ -0,0 +1,47 @@
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/tcp.h>
+#include <netinet/in.h>
+#include <linux/types.h>
+
+#include "zdtmtst.h"
+
+const char *test_doc = "Check both dump and restore with tcp_close";
+const char *test_author = "Bui Quang Minh <minhquangbui99@gmail.com>";
+
+static int port = 8880;
+
+int main(int argc, char **argv)
+{
+	int fd_s, fd, client;
+
+	test_init(argc, argv);
+
+	fd_s = tcp_init_server(AF_INET, &port);
+	if (fd_s < 0) {
+		pr_err("Server initializations failed\n");
+		return 1;
+	}
+
+	client = tcp_init_client(AF_INET, "localhost", port);
+	if (client < 0) {
+		pr_err("Client initializations failed\n");
+		return 1;
+	}
+
+	fd = tcp_accept_server(fd_s);
+	if (fd < 0) {
+		pr_err("Can't accept client\n");
+		return 1;
+	}
+	close(fd_s);
+
+	shutdown(client, SHUT_WR);
+	shutdown(fd, SHUT_WR);
```

Comment:
why do we need to shutdown both sockets?

Reply:
This is for making sockets into TCP_CLOSE state but still have `dst_port` for reaching `restore_one_tcp`

Reply:
Should we have another pair of sockets that are not closed?

Reply:
The other cases when TCP sockets are in connected or listening state are covered in your `socket-dump-tcp-close.c` test. Maybe we can merge these 2 into 1 test case.